### PR TITLE
Payment history CSV receipt links #3477 (WIP)

### PR DESCRIPTION
**Pull request by willy-b**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist. (issue #3477)

- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.

- [ ] Added/updated tests for this change (for new code or code which already has tests)

**^^ tests on the way pending general review of this PR**

- [ ] Ran `git rebase -i` to squash commits (if needed).

^^ will probably squash after adding tests

Test Plan:
Needs tests. Not sure if CSV export is desired in interim and so getting that feedback while writing tests to save time.

### CODE REVIEW

**Comment by willy-b** [comment]
depends on a PR in ledger-client: https://github.com/brave/ledger-client/pull/15

File: app/ledger.js [hunk 1]
```diff
@@ -286,6 +286,25 @@ if (ipc) {
     if (balanceTimeoutId) clearTimeout(balanceTimeoutId)
     balanceTimeoutId = setTimeout(getBalance, 5 * msecs.second)
   })
+
+  ipc.on(messages.OPEN_LEDGER_TRANSACTION_CSV, (event, viewingIds, csvFilename) => {
+    if (client) {
+      var txCsvText = client._getTransactionCSVText(viewingIds)
```
Comment (diracdeltas) [inline app/ledger.js]: i would replace all of the below with `win.webContents.downloadURL(url)`, where `url` is `txCsvText` as a Blob object URL or perhaps a data URL. then the CSV file will be automatically written to the default downloads folder, and the user can click on the downloads bar to open it.

File: app/ledger.js [hunk 2]
```diff
@@ -286,6 +287,18 @@ if (ipc) {
...
   })
+
+  ipc.on(messages.OPEN_LEDGER_TRANSACTION_CSV, (event, viewingIds, csvFilename) => {
+    if (client) {
+      let txCsvText = client._getTransactionCSVText(viewingIds)
+      let txCsvTextDataURI = 'data:text/csv;base64,' + base64Encode(txCsvText)
+
+      const win = electron.BrowserWindow.getFocusedWindow()
+      if (win && win.webContents) {
+        win.webContents.downloadURL(txCsvTextDataURI)
```
Comment (diracdeltas) [inline app/ledger.js]: unfortunately there seems to be no way to specify the filename so it defaults to `download.csv`. but i'm fine with merging this as-is and fixing the filename in a later issue. thanks!

**Comment by diracdeltas** [comment]
@mrose17 this looks good to me to merge once ledger-client is updated.

**Comment by mrose17** [comment]
@willy-b i'm going to merge https://github.com/brave/ledger-client/pull/15 and make a few changes, then push the new version.

@diracdeltas - i will ping you when that is done so you can merge this PR. thanks!

**Comment by mrose17** [comment]
@diracdeltas - i have published ledger-client@0.8.78 - please update and commit as you see fit. thanks!

**Comment by diracdeltas** [comment]
fixing and rebasing this PR now

**Comment by mrose17** [comment]
+1 thanks!

**Comment by diracdeltas** [comment]
cherry-picked https://github.com/brave/browser-laptop/commit/7c44c28d79088af4a24a62057b4eb2d22cf42c2c

**Comment by willy-b** [comment]
thanks @diracdeltas & @mrose17!

**Pull request closed without merging**